conans/util/files.py

```diff
@@ -161,10 +161,8 @@ def _generic_algorithm_sum(file_path, algorithm_name):
 
 
 def save_append(path, content, encoding="utf-8"):
-    try:
-        os.makedirs(os.path.dirname(path))
-    except Exception:
-        pass
+    if os.path.dirname(path) != '':
```

Comment:
Try this:

```python
dirname = os.path.dirname(path)
if dirname and not os.path.exists(dirname):  # compat Python 2.7, otherwise exists_ok=True
    try:
       os.makedirs(dirname)
    except Exception as e:
       raise ConanException("Couldn't create folder '%s': %s" % (dirname, str(e))

```

Reply:
The `mkdir` function already checks for the existence of the folder. So I don't think we need to do that here.

Reply:
Updated the PR. I took the liberty of using the format function instead of the old string interpolation style. And like I said, I didn't include the check for existence of the folder, so it will be done in `mkdir`.

What do you think of doing all this in mkdir? Having two copies of the same code in `save` and `save_append` really bugs me, but I don't know how changing `mkdir` can affect the rest of the codebase.